compiler/dalgona/include/PostOperatorHook.h

```diff
@@ -0,0 +1,164 @@
+#include <luci_interpreter/core/Tensor.h>
+#include <luci/IR/CircleNodeVisitor.h>
+#include <luci/IR/CircleNodeDecl.h>
+
+#include <pybind11/embed.h>
+#include <pybind11/numpy.h>
+#include <vector>
+#include <unordered_map>
+
+using Tensor = luci_interpreter::Tensor;
+
+namespace py = pybind11;
+using namespace py::literals;
+
+namespace dalgona
+{
+
+class PostOperatorHook final : public luci::CircleNodeVisitor<void>
+{
+
+// This macro describes common tasks inside post-opeartor hook
+// Use this macro for hooks for normal Ops (except DefaultOp)
+// After this macro, three variables are created
+// 1. hook: Python function called after operator execution (type: py::object)
+// 2. inputs: input data (type: std::vector of numpy array)
+// 3. output: output data (type: numpy array)
+#define POST_OPERATOR_HOOK_PROLOGUE(OP_NAME)            \
+  if (!py::hasattr(_analysis, #OP_NAME "Post"))         \
+  {                                                     \
+    visit(static_cast<const luci::CircleNode *>(node)); \
+    return;                                             \
+  }                                                     \
+  py::object hook = _analysis.attr(#OP_NAME "Post");    \
+  auto inputs = inputsPyArray(node, _interpreter);      \
+  auto output = outputPyArray(node, _interpreter);
+
+private:
+  py::object _analysis;
+  luci_interpreter::Interpreter *_interpreter;
+
+public:
+  explicit PostOperatorHook(py::object analysis, luci_interpreter::Interpreter *interpreter)
+      : _analysis(analysis), _interpreter(interpreter)
+  {
+    // Do nothing
+  }
+
+  // default
+  void visit(const luci::CircleNode *node)
+  {
+    if (!py::hasattr(_analysis, "DefaultOpPost"))
+      return;
+
+    py::object hook = _analysis.attr("DefaultOpPost");
+    auto inputs = inputsPyArray(node, _interpreter);
+    auto output = outputPyArray(node, _interpreter);
+
+    py::list input_list;
+    for (int i = 0; i < inputs.size(); i++)
+    {
+      input_list.append(inputs[i]);
+    }
+
+    hook(node->name(),             // name
+         toString(node->opcode()), // opcode
+         input_list,               // list of inputs
+         output                    // output
+         );
+  }
+
+  void visit(const luci::CircleConv2D *node)
+  {
+    POST_OPERATOR_HOOK_PROLOGUE(Conv2D)
+
+    auto padding = node->padding();
+    auto stride = node->stride();
+    auto dilation = node->dilation();
+
+    auto py_stride = py::dict("w"_a = stride->w(), "h"_a = stride->h());
+    auto py_dilation = py::dict("w"_a = dilation->w(), "h"_a = dilation->h());
+
+    auto fused_act = node->fusedActivationFunction();
+
+    hook(node->name(),                                      // name
+         inputs[0],                                         // input
+         inputs[1],                                         // filter
+         inputs[2],                                         // bias
+         padding == luci::Padding::SAME ? "SAME" : "VALID", // padding
```

Comment:
(optional) use `padding_str = padding == luci::Padding::SAME ? "SAME" : "VALID"` so that arguments and explanation comments get closer.